Add February 2021 Acidanthera Post (#16)
dortania/dortania.github.io: _posts/2021-02-01-acidanthera-february.md

---
layout: post
title:  "Acidanthera Updates: February 2021"
date:   2021-02-01 15:00:00 -0700
categories: Hackintosh updates
---

* [A message from vit9696](#a-message-from-vit9696)
* [Dortania Updates](#dortania-updates)
* [Changelogs](#changelogs)

# A message from vit9696

It is nice to meet you with the first actually made in 2021 release. This time we have a handful changes to share with.
 

> As some of you noticed this time we slightly postponed the deadline from the usual, and the primary reason behind this was to be able to include some imminent changes in OpenCore. After a long and tiring work with the PE/COFF loader [@mhaeuser](https://github.com/mhaeuser) (ex Download-Fritz) finally wrote a formal proof of its most critical properties. As we plan to later include it in EDK II and essentially in every new UEFI firmware, we now launch a second testing phase by embedding the up-to-date version in OpenCore. For the bootloader the most important changes are the fixes for certain potential memory corruptions and enhanced APFS driver signature verification, provided as an extension to the base code. Our technically inclined users may also find the [original paper](https://github.com/mhaeuser/ISPRASOpen-SecurePE) to be an interesting read.

Reviewer: ```suggestion
As some of you noticed this time we slightly postponed the deadline from the usual, and the primary reason behind this was to be able to include some imminent changes in OpenCore. After some long and tiring work with the PE/COFF loader [@mhaeuser](https://github.com/mhaeuser) (ex Download-Fritz) finally wrote a formal proof of its most critical properties. As we plan to later include it in EDK II and essentially in every new UEFI firmware, we now launch a second testing phase by embedding the up-to-date version in OpenCore. For the bootloader, the most important changes are the fixes for certain potential memory corruptions and enhanced APFS driver signature verification, provided as an extension to the base code. Our technically-inclined users may also find the [original paper](https://github.com/mhaeuser/ISPRASOpen-SecurePE) to be an interesting read.
```

> The only more important thing to security is what the end-user interacts with, at least for some :P. This is why [@mhaeuser](https://github.com/mhaeuser) continued his OpenCanopy project introducing item scrolling with a large amount of boot options, background image support, which can also be used as a logo through transparency and background colour, and relaxed image dimensions, which of course are well-documented in the PDF spec following our usual standards. Marginal usability improvements also cover [@PMheart](https://github.com/PMheart) additions to ocvalidate, which many found really helpful, and [@GoldFish](https://github.com/GoldFish) audio configuration additions to SysReport. Besides that, we got two new drivers: OpenHfsPlus and OpenPartitionDxe. OpenHfsPlus is a tireless effort by [@shchuko](https://github.com/shchuko), who continued the [@gsomlo](https://github.com/gsomlo) and [@nms42](https://github.com/nms42) work, by finishing a poison-free non-GPL HFS+ driver. While it still is inferior to Apple driver, its speed is clearly better than VBoxHfs, which is now deprecated. OpenPartitionDxe on the other side improves compatibility with legacy macOS recoveries.

Reviewer: ```suggestion
The only more important thing to security is what the end-user interacts with, at least for some :P. This is why [@mhaeuser](https://github.com/mhaeuser) continued his OpenCanopy project introducing item scrolling with a large amount of boot options, background image support, which can also be used as a logo through transparency and background colour, and relaxed image dimensions, which of course are well-documented in the PDF spec following our usual standards. Marginal usability improvements also cover [@PMheart's](https://github.com/PMheart) additions to ocvalidate, which many found really helpful, and [@Goldfish64's](https://github.com/Goldfish64) audio configuration additions to SysReport. Besides that, we got two new drivers: OpenHfsPlus and OpenPartitionDxe. OpenHfsPlus is a tireless effort by [@shchuko](https://github.com/shchuko), who continued [@gsomlo's](https://github.com/gsomlo) and [@nms42's](https://github.com/nms42) work by finishing a poison-free non-GPL HFS+ driver. While it is still inferior to Apple's driver, its speed is clearly better than VBoxHfs, which is now deprecated. OpenPartitionDxe on the other hand improves compatibility with legacy macOS recoveries.
```

> The only more important thing than the security is what the end-user interacts with, at least for some :P. This is why [@mhaeuser](https://github.com/mhaeuser) continued his OpenCanopy project introducing item scrolling for a large amount of boot options, background image support, which can also be used as a logo through transparency and background colour, and relaxed image dimensions, which of course are well-documented in the PDF spec following our usual standards. Marginal usability improvements also cover [@PMheart](https://github.com/PMheart) additions to ocvalidate, which many found really helpful, and [@GoldFish](https://github.com/GoldFish) audio configuration additions to SysReport. Besides that, we got two new drivers: OpenHfsPlus and OpenPartitionDxe. OpenHfsPlus is a tireless effort by [@shchuko](https://github.com/shchuko), who continued the [@gsomlo](https://github.com/gsomlo) and [@nms42](https://github.com/nms42) work, by finishing a poison-free non-GPL HFS+ driver. While it still is inferior to Apple driver, its speed is clearly better than VBoxHfs, which is now deprecated. OpenPartitionDxe on the other side improves compatibility with legacy macOS recoveries.

Reviewer: ```suggestion
The only more important thing than the security is what the end-user interacts with, at least for some :P. This is why [@mhaeuser](https://github.com/mhaeuser) continued his OpenCanopy project introducing item scrolling for a large amount of boot options, background image support, which can also be used as a logo through transparency and background colour, and relaxed image dimensions, which of course are well-documented in the PDF spec following our usual standards. Marginal usability improvements also cover [@PMheart](https://github.com/PMheart) additions to ocvalidate, which many found really helpful, and [@Goldfish64](https://github.com/Goldfish64) audio configuration additions to SysReport. Besides that, we got two new drivers: OpenHfsPlus and OpenPartitionDxe. OpenHfsPlus is a tireless effort by [@shchuko](https://github.com/shchuko), who continued [@gsomlo's](https://github.com/gsomlo) and [@nms42's](https://github.com/nms42) work, by finishing a poison-free non-GPL HFS+ driver. While it still is inferior to Apple's driver, its speed is clearly better than VBoxHfs, which is now deprecated. OpenPartitionDxe on the other hand improves compatibility with legacy macOS recoveries.
```

> Sometimes the changes are not dictated by the necessity to improve things on our end or even our bugs, but rather flaws in the design & code of the other parties. This release includes three changes that happened due to exactly this kind of issues. One of them is the result of Apple’s APFS design, which does not track trimmed areas, resulting in essentially broken trim support for a vast amount of third-party SSDs. The discovery and an ugly workaround are described in the [SetApfsTrimTimeout section](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf) thanks to [@lvs1974's](https://github.com/lvs1974) efforts. The second issue is a design flaw of the older SMBIOS specifications, which caused confusion of how UUIDs can be read and written. While no changes are needed for the existing users, some new users might benefit from the improvements in dmidecode and the new [SetApfsTrimTimeout option](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf). Last but not least was a discovery that some firmwares disable the support for UEFI driver loading, even with UEFI Secure Boot disabled. This obviously violates the UEFI specification, but [Microsoft](https://github.com/badstorm/surface-pro-7-opencore/issues/9) thinks differently, breaking the entire loading of OpenCore. If any Microsoft Surface firmware developers are reading this, it will help if you provide a comment on the matter, since even the provided workaround by transitioning to the app only uncovered further issues.

Reviewer: ```suggestion
Sometimes the changes are not dictated by the necessity to improve things on our end or even our bugs, but rather flaws in the design & code of the other parties. This release includes three changes that happened due to exactly this kind of issues. One of them is the result of Apple’s APFS design, which does not track trimmed areas, resulting in essentially broken trim support for a vast amount of third-party SSDs. The discovery and an ugly workaround are described in the [SetApfsTrimTimeout section](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf) thanks to [@lvs1974's](https://github.com/lvs1974) efforts. The second issue is a design flaw of the older SMBIOS specifications, which caused confusion of how UUIDs can be read and written. While no changes are needed for the existing users, some new users might benefit from the improvements in dmidecode and the new [SetApfsTrimTimeout option](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf). Last but not least was a discovery that some firmwares disable support for UEFI driver loading, even with UEFI Secure Boot disabled. This obviously violates the UEFI specification, but [Microsoft](https://github.com/badstorm/surface-pro-7-opencore/issues/9) thinks differently, breaking the entire loading of OpenCore. If any Microsoft Surface firmware developers are reading this, it will help if you provide a comment on the matter, since even the provided workaround by transitioning to the app only uncovered further issues.
```

> Sometimes the changes are not dictated by the necessity to improve things on our end or even our bugs, but rather flaws in the design & code of the other parties. This release includes three changes that happened due to exactly this kind of issues. One of them is the result of Apple’s APFS design, which does not track trimmed areas, resulting in essentially broken trim support for a vast amount of third-party SSDs. The discovery and an ugly workaround are described in [SetApfsTrimTimeout section](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf) thanks to [@lvs1974](https://github.com/lvs1974) efforts. The second issue is a design flaw of the older SMBIOS specifications, which caused confusion of how UUIDs can be read and written. While no changes are needed for the existing users, some new users might benefit from the improvements in dmidecode and the new [SetApfsTrimTimeout option](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf). Last but not least was a discovery that some firmwares disable the support for UEFI driver loading, even with UEFI Secure Boot disabled. This obviously violates the UEFI specification, but [Microsoft](https://github.com/badstorm/surface-pro-7-opencore/issues/9) thinks differently, breaking the entire loading of OpenCore. If any of Microsoft Surface firmware developers are reading it, it will help if you provide a comment on the matter, since even the provided workarounds by transitioning to the app, it only uncovered further issues.

Reviewer: ```suggestion
Sometimes the changes are not dictated by the necessity to improve things on our end or even our bugs, but rather flaws in the design & code of the other parties. This release includes three changes that happened due to exactly this kind of issues. One of them is the result of Apple’s APFS design, which does not track trimmed areas, resulting in essentially broken trim support for a vast amount of third-party SSDs. The discovery and an ugly workaround are described in the [SetApfsTrimTimeout section](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf) thanks to [@lvs1974's](https://github.com/lvs1974) efforts. The second issue is a design flaw of the older SMBIOS specifications, which caused confusion of how UUIDs can be read and written. While no changes are needed for the existing users, some new users might benefit from the improvements in dmidecode and the new [SetApfsTrimTimeout option](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf). Last but not least was a discovery that some firmwares disable the support for UEFI driver loading, even with UEFI Secure Boot disabled. This obviously violates the UEFI specification, but [Microsoft](https://github.com/badstorm/surface-pro-7-opencore/issues/9) thinks differently, breaking the entire loading of OpenCore. If any Microsoft Surface firmware developers are reading this, it will help if you provide a comment on the matter, since even the provided workaround by transitioning to the app only uncovered further issues.
```

> Besides OpenCore, our kexts also got notable updates. AppleALC got legacy Mac support, thanks to [@khronokernel](https://github.com/khronokernel), SMCDellSensors got fan control improvements, thanks to to [@lvs1974](https://github.com/lvs1974), and on the Lilu side we added a few helper APIs that show how one can perform basic userspace patches in Big Sur. The usage of this was added to WhateverGreen to improve MacPro5,1 compatibility. This makes it all for today; we are all very glad to hear your support.

Reviewer: ```suggestion
Besides OpenCore, our kexts also got notable updates. AppleALC got legacy Mac support, thanks to [@khronokernel](https://github.com/khronokernel), SMCDellSensors got fan control improvements, thanks to [@lvs1974](https://github.com/lvs1974), and on the Lilu side, we added a few helper APIs that show how one can perform basic userspace patches in Big Sur. This was used in WhateverGreen to improve MacPro5,1 compatibility. This makes it all for today; we are all very glad to hear your support.
```

> Besides OpenCore, our kexts also got notable updates. AppleALC got legacy Mac support, thanks to [@khronokernel](https://github.com/khronokernel), SMCDellSensors got fan control improvements, thanks to to [@lvs1974](https://github.com/lvs1974), and on Lilu side we added a few helper APIs that show how one can perform basic userspace patches in Big Sur. The usage of this was added to WhateverGreen to improve MacPro5,1 compatibility. This makes it for today, we are all very glad to hear your support.

Reviewer: ```suggestion
Besides OpenCore, our kexts also got notable updates. AppleALC got legacy Mac support, thanks to [@khronokernel](https://github.com/khronokernel), SMCDellSensors got fan control improvements, thanks to to [@lvs1974](https://github.com/lvs1974), and on the Lilu side we added a few helper APIs that show how one can perform basic userspace patches in Big Sur. The usage of this was added to WhateverGreen to improve MacPro5,1 compatibility. This makes it all for today; we are all very glad to hear your support.
```
